pkg/resources/task.go

```diff
@@ -0,0 +1,645 @@
+	var (
+		database = data.Get("database").(string)
+		dbSchema = data.Get("schema").(string)
+		name     = data.Get("name").(string)
+		after    = data.Get("after").(string)
+	)
+
+	if name == "" {
+		return nil, nil
+	}
+
+	// always start from first predecessor
+	// or the current task when standalone
+	if after != "" {
+		name = after
+	}
+
+	for {
+
+		builder := snowflake.Task(name, database, dbSchema)
+		q := builder.Show()
+		row := snowflake.QueryRow(db, q)
+		task, err := snowflake.ScanTask(row)
+
+		if err != nil && name != data.Get("name").(string) {
+			return nil, errors.Wrapf(err, "failed to locate the root node of: %v", name)
+		}
+
+		if task.Predecessors == nil {
+			log.Println(fmt.Sprintf("[DEBUG] found root task: %v", name))
+			// we only want to deal with suspending the root task when its enabled (started)
+			if task.IsEnabled() {
+				return snowflake.Task(name, database, dbSchema), nil
+			}
+			return nil, nil
+		}
+
+		name = task.GetPredecessorName()
+	}
+}
+
+// getRootTaskAndSuspend retrieves the root task and suspends it
+func getRootTaskAndSuspend(data *schema.ResourceData, meta interface{}) (*snowflake.TaskBuilder, error) {
+	db := meta.(*sql.DB)
+	name := data.Get("name").(string)
+
+	root, err := getRootTask(data, meta)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error retrieving root task %v", name)
+	}
+
+	if root != nil {
+		qr := root.Suspend()
+		err = snowflake.Exec(db, qr)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error suspending root task %v", name)
+		}
+	}
+
+	return root, nil
+}
+
+func resumeTask(root *snowflake.TaskBuilder, meta interface{}) {
+	if root == nil {
+		return
+	}
+
+	db := meta.(*sql.DB)
+	qr := root.Resume()
+	err := snowflake.Exec(db, qr)
+	if err != nil {
+		log.Fatal(errors.Wrapf(err, "error resuming root task %v", root.QualifiedName()))
+	}
+}
+
+// taskIDFromString() takes in a pipe-delimited string: DatabaseName|SchemaName|TaskName
+// and returns a taskID object
+func taskIDFromString(stringID string) (*taskID, error) {
+	reader := csv.NewReader(strings.NewReader(stringID))
+	reader.Comma = pipeIDDelimiter
+	lines, err := reader.ReadAll()
+	if err != nil {
+		return nil, fmt.Errorf("Not CSV compatible")
+	}
+
+	if len(lines) != 1 {
+		return nil, fmt.Errorf("1 line per task")
+	}
+	if len(lines[0]) != 3 {
+		return nil, fmt.Errorf("3 fields allowed")
+	}
+
+	taskResult := &taskID{
+		DatabaseName: lines[0][0],
+		SchemaName:   lines[0][1],
+		TaskName:     lines[0][2],
+	}
+	return taskResult, nil
+}
+
+// Task returns a pointer to the resource representing a task
+func Task() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateTask,
+		Read:   ReadTask,
+		Update: UpdateTask,
+		Delete: DeleteTask,
+		Exists: TaskExists,
+
+		Schema: taskSchema,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+// ReadTask implements schema.ReadFunc
+func ReadTask(data *schema.ResourceData, meta interface{}) error {
```

Comment:
>>>What do you mean by not reading state? Thanks!

Here is ReadTask you need to read the parameters to be able to add them to the state

Reply:
@chriskuchin does the `SHOW TASKS` return the `session_parameters`?

Reply:
no you have to do another query you can see in my PR. It's a show parameters query